src/ray/common/id.h

```diff
@@ -27,6 +27,67 @@ class JobID;
 /// A helper function that get the `DriverID` of the given job.
 WorkerID ComputeDriverIdFromJob(const JobID &job_id);
 
+enum class ObjectType : uint8_t {
+  PUT_OBJECT = 0x0,
+  RETURN_OBJECT = 0x1,
+};
+
+// TODO(qwang): Unify this definition with `core_worker`.
+enum class TransportType : uint8_t {
+  STANDARD = 0x0,
+  DIRECT_ACTOR_CALL = 0x1,
+};
+
+using ObjectIDFlagsType = uint16_t;
+using ObjectIDIndexType = uint32_t;
+
+namespace {
+
+constexpr uint8_t kIsTaskBitsOffset = 15;
+constexpr uint8_t kObjectTypeBitsOffset = 14;
+constexpr uint8_t kTransportTypeBitsOffset = 11;
+
+constexpr ObjectIDFlagsType kIsTaskFlagBitMask =  0x1 << kIsTaskBitsOffset;
+constexpr ObjectIDFlagsType kObjectTypeFlagBitMask = 0x1 << kObjectTypeBitsOffset;
+constexpr ObjectIDFlagsType kTransportTypeFlagBitMask = 0x7 << kTransportTypeBitsOffset;
+
+/// A helper function to set object ids flags.
+inline void SetIsTaskFlag(bool is_task, ObjectIDFlagsType *flags) {
+  ObjectIDFlagsType is_task_bits;
+  if (is_task) {
+    is_task_bits = (0x1 << kIsTaskBitsOffset);
+  } else {
+    is_task_bits = (0x0 << kIsTaskBitsOffset);
+  }
+  *flags = (*flags bitor is_task_bits);
+}
+
+inline void SetObjectTypeFlag(ObjectType object_type, ObjectIDFlagsType *flags) {
+  const ObjectIDFlagsType object_type_bits = static_cast<ObjectIDFlagsType>(object_type) << kObjectTypeBitsOffset;
+  *flags = (*flags bitor object_type_bits);
+}
+
+inline void SetTransportTypeFlag(TransportType transport_type, ObjectIDFlagsType *flags) {
+  const ObjectIDFlagsType transport_type_bits = static_cast<ObjectIDFlagsType>(transport_type) << kTransportTypeBitsOffset;
+  *flags = (*flags bitor transport_type_bits);
+}
+
+inline bool IsTask(ObjectIDFlagsType flags) {
+  return ((flags bitand kIsTaskFlagBitMask) >> kIsTaskBitsOffset) != 0x0;
+}
+
+inline ObjectType GetObjectType(ObjectIDFlagsType flags) {
+  const ObjectIDFlagsType object_type = (flags bitand kObjectTypeFlagBitMask) >> kObjectTypeBitsOffset;
+  return static_cast<ObjectType>(object_type);
+}
+
+inline TransportType GetTransportType(ObjectIDFlagsType flags) {
+  const ObjectIDFlagsType transport_type = (flags bitand kTransportTypeFlagBitMask) >> kTransportTypeBitsOffset;
+  return static_cast<TransportType>(transport_type);
+}
```

Comment:
Can these helper functions be moved to the cc file? they're not used in this header file.

Reply:
But there's a test depends on these methods.
Whatever we should declare these methods in the header file at least. Otherwise, we must write the tests in `id.cc` file.